tutorials/embedded-c-getting-started/index.md

```diff
@@ -0,0 +1,297 @@
+```
+#ifdef __cplusplus
+
+extern "C" {
+  #endif
+
+uint8_t temprature_sens_read();
+
+#ifdef __cplusplus
+}
+#endif
+```
+
+### Google Cloud IoT Core
+
+If you’ve never used IoT Core, don’t worry, the steps below will get you setup to transmit telemetry data to the cloud but before we can do that lets talk about IoT Core and its components. Google Cloud IoT Core is a complete set of tools to connect, process, store, and analyze data both at the edge and in the cloud. Google Cloud IoT consists of the device management API for creating and managing logical collections of devices and the protocol bridge which adapts device-friendly protocols (MQTT or HTTP) to scalable Google infrastructure.
+
+Now that we have a little bit of information about Google Cloud IoT Core lets set it up.
+
+To learn more about the protocols for [Google Cloud IoT Core](https://cloud.google.com/iot/docs/), read the [MQTT](https://cloud.google.com/iot/docs/how-tos/mqtt-bridge) and [HTTP](https://cloud.google.com/iot/docs/how-tos/http-bridge) documentation.
+
+### Setting up your device registry
+
+Before connecting to Google Cloud you need to create device authorization credentials and a device registry to contain your devices.
+
+There are two ways you can set up your project on Google Cloud IoT, you can use the Cloud SDK (gcloud) or using the UI in the [Google Cloud Console](https://console.cloud.google.com/) This guide will go through setting the project up using gcloud. After you have downloaded the [Cloud SDK](https://cloud.google.com/sdk).
+
+**To set up your device registry :**
+1. Generate Elliptic Curve (EC) device credentials for authenticating the device when it’s
+trying to connect with the cloud, You will need to know where these files are later so make sure they’re saved somewhere you can access.
+```
+openssl ecparam -genkey -name prime256v1 -noout -out ec_private.pem openssl ec -in ec_private.pem -pubout -out ec_public.pem
+```
+1. Make sure your gcloud is up to date. gcloud components update
+1. Create a PubSub topic and subscription used for storing telemetry .
+```
+gcloud pubsub topics create temperature
+gcloud pubsub subscriptions create data --topic=temperature
+```
+1. Create a device registry and add a device to the registry .
+```
+gcloud iot registries create esp-test --region=us-central1 --event-notification-config=topic=temperature gcloud iot devices create test-dev --region=us-central1 --registry=esp-test \ --public-key path=ec_public.pem,type=es256
+```
+
+### Cloning mqtt example
+
+You will need to clone the repo to get the example code, in your terminal go to a location you want to store the cloned repo and run the following command:
+
+``git clone https://github.com/espressif/esp-google-iot --recurse-submodules``
+
+Recuse submodules is important because you will need the IoT Core Embedded C SDK which is included in the repository as a submodule.
```

Comment:
```suggestion
Recurse submodules is important because you will need the IoT Core Embedded C SDK which is included in the repository as a submodule.
```